View3D/Services/SimplygonHelpers.cs

```diff
@@ -0,0 +1,190 @@
+            }
+        }
+        public static void FillSGTriangles(ref spGeometryData geometryDataSG, MeshObject originalMesh)
+        {
+            // -- construct triangles, and add values for polygon corners
+            for (int cornerIndex = 0; cornerIndex < originalMesh.IndexArray.Length; cornerIndex++) 
+            {                
+                ushort vertexIndex = originalMesh.IndexArray[cornerIndex];
+
+                // set vertex index for the triangle corner
+                geometryDataSG.GetVertexIds().SetItem(cornerIndex, vertexIndex);
+
+                ref var sourceVertex = ref originalMesh.VertexArray[vertexIndex];
+
+                // -- set tex_coords channel 1
+                var TextureCoords1 = geometryDataSG.GetTexCoords(0);
+                TextureCoords1.SetTupleSize(2);
+                TextureCoords1.SetTuple(cornerIndex, new float[] { sourceVertex.TextureCoordinate.X, sourceVertex.TextureCoordinate.Y });
+
+                // -- set tex_coords channel 2 (empty in Ole's "commom format")
+                // TODO: extend "common format" to having 2 UV channels, as many/some "default/static" models use that
+                var textureCoords2 = geometryDataSG.GetTexCoords(1);
+                textureCoords2.SetTupleSize(2);
+                textureCoords2.SetTuple(cornerIndex, new float[] { 0, 0 });
+
+                // -- Set normals
+                var normals = geometryDataSG.GetNormals();
+                normals.SetTupleSize(3);
+                normals.SetTuple(cornerIndex, new float[] { sourceVertex.Normal.X, sourceVertex.Normal.Y, sourceVertex.Normal.Z });
+
+                // -- Set tangents
+                var tangents = geometryDataSG.GetTangents(0);
+                tangents.SetTupleSize(3);
+                tangents.SetTuple(cornerIndex, new float[] { sourceVertex.Tangent.X, sourceVertex.Tangent.Y, sourceVertex.Tangent.Z });
+
+                // -- Set bitangents
+                var bitangents = geometryDataSG.GetBitangents(0);
+                bitangents.SetTupleSize(3);
+                bitangents.SetTuple(cornerIndex, new float[] { sourceVertex.BiNormal.X, sourceVertex.BiNormal.Y, sourceVertex.BiNormal.Z });
+            }
+        }                
+        public static void ReduceSGMesh(ISimplygon sg, spGeometryData pGeometryData, out spPackedGeometryData packedGeometryDataSG, float factor)
+        {
+            // -- create scene graph and add mesh
+            var sgScene = sg.CreateScene();
+            var mesh = sg.CreateSceneMesh();
+            mesh.SetGeometry(pGeometryData);
+            sgScene.GetRootNode().AddChild(mesh);
+            
+            // -- create SG mesh reduction processor, which only exists in the following scope, and thus is cleaned up at the end
+            using (var sgReductionProcessor = sg.CreateReductionProcessor())
```

Comment:
Do: 

using var sgReductionProcessor = sg.CreateReductionProcessor();

This way one does net the {}.